### PR TITLE
Accounting time tracking locked create

### PR DESCRIPTION
**Pull request by sajberkurajber90**

### CODE REVIEW

**Comment by sajberkurajber90** [comment]
when pause takes longer then start and end - error is trown - need validation

File: src/Actions/WorkTime/CreateLockedWorkTime.php [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+    public function performAction(): WorkTime
+    {
+        $workTime = app(WorkTime::class, ['attributes' => $this->data]);
+
+        if (is_null(data_get($this->data, 'is_billable'))) {
+            $workTime->is_billable = $workTime->workTimeType?->is_billable ?? false;
+        }
+
+       if (is_null($workTime->paused_time_ms)) {
+            $workTime->paused_time_ms = 0;
```
Comment (patrickweh) [inline src/Actions/WorkTime/CreateLockedWorkTime.php:26]: ??=

File: src/Actions/WorkTime/CreateLockedWorkTime.php [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+        parent::boot($data);
+        $this->rules = resolve_static(CreateLockedWorkTimeRuleset::class, 'getRules');
+    }
+
+    public function performAction(): WorkTime
+    {
+        $workTime = app(WorkTime::class, ['attributes' => $this->data]);
+
+        if (is_null(data_get($this->data, 'is_billable'))) {
+            $workTime->is_billable = $workTime->workTimeType?->is_billable ?? false;
```
Comment (patrickweh) [inline src/Actions/WorkTime/CreateLockedWorkTime.php:22]: ??=

File: src/Actions/WorkTime/CreateLockedWorkTime.php [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+        return $workTime->fresh();
+    }
+
+    protected function prepareForValidation(): void
+    {
+        $this->data['user_id'] ??= auth()->user()->id;
+
+        if (! is_null($this->data['user_id']) && ! is_null($this->data['started_at'])) {
+            // add parent_id in case daily work time exists
+            $this->data['parent_id'] = WorkTime::query()->where('user_id', $this->data['user_id'])
```
Comment (patrickweh) [inline src/Actions/WorkTime/CreateLockedWorkTime.php:43]: app(Worktime::class)->query()

File: src/Rulesets/WorkTime/CreateLockedWorkTimeRuleset.php [hunk 4]
```diff
@@ -0,0 +1,23 @@
+<?php
+
+namespace FluxErp\Rulesets\WorkTime;
+
+use FluxErp\Models\WorkTime;
+use FluxErp\Rules\ModelExists;
+
+class CreateLockedWorkTimeRuleset extends CreateWorkTimeRuleset
+{
+    public function rules(): array
```
Comment (patrickweh) [inline src/Rulesets/WorkTime/CreateLockedWorkTimeRuleset.php:10]: Use array merge with parent::rules() and the new rules

File: tests/Browser/OrdersTest.php [hunk 5]
```diff
@@ -25,4 +25,21 @@ public function test_user_can_see_orders()
                 ->waitForText(text: 'Order Type -> Name', seconds: 30);
         });
     }
+
+    // test for deleting order
+    public function test_user_can_delete_open_order()
```
Comment (patrickweh) [inline tests/Browser/OrdersTest.php:30]: I think this is not finished?

**Review by patrickweh — CHANGES_REQUESTED** [review]

File: tests/Browser/OrdersTest.php (hunk 5 above)
Comment (sajberkurajber90) [reply tests/Browser/OrdersTest.php:30]: this doesent belong to this branch - how it ended up here - maybe i merged test - browser branch

File: src/Actions/WorkTime/CreateLockedWorkTime.php (hunk 1 above)
Comment (sajberkurajber90) [reply src/Actions/WorkTime/CreateLockedWorkTime.php]: moved to prepareForValidation

**Review by SirSplasch — APPROVED** [review]

**Pull request merged**